### DISCUSSION TITLE
Compile template tag to hyperscript

### DISCUSSION THREAD

**Discussion by marcodpt** [MithrilJS/mithril.js#2803] [Show and tell]
I created a view engine based on the DOM template tag, which compiles to hyperscript functions:

[tint](https://github.com/marcodpt/tint)

See for example what a simple `TODO` app looks like:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <title>mithril + tint</title>
    <link rel="shortcut icon" type="image/x-icon" href="favicon.ico">
    <script src="https://unpkg.com/mithril/mithril.js"></script>
    <script type="module">
      import tint from 'https://cdn.jsdelivr.net/gh/marcodpt/tint/index.js'
      const render = tint(m)

      const state = {
        todos: [],
        value: ""
      }
      const AddTodo = () => {
        state.todos.push(state.value)
        state.value = ""
        rerender()
      }
      const NewValue = (ev) => {
        state.value = ev.target.value
      }
      const rerender = () => {
        const root = document.getElementById("app")
        m.render(root, render('my-todo', {
          ...state,
          AddTodo,
          NewValue
        }, root.tagName, {id: 'app'}))
      }

      rerender()
    </script>
  </head>
  <body>
    <main id="app"></main>
    <template id="my-todo">
      <h1>To do list</h1>
      <input type="text" :value="value" :oninput="NewValue">
      <ul>
        <li :each="todos" :text></li>
      </ul>
      <button :onclick="AddTodo">New!</button>
    </template>
  </body>
</html>
```

[LIVE PREVIEW](https://marcodpt.github.io/tint/mithril.html)

In this example, what was achieved was the total separation of the javascript layouts, without using build steps, and only introducing as a dependency a library with ~130 lines of javascript code.

Which I find extremely useful for maintaining large projects, especially working with designers who only have knowledge of css and html.

Another great feat that was achieved is that in the examples I build the same app:
 - [without any framework](https://raw.githubusercontent.com/marcodpt/tint/main/todo.html)
 - [preact](https://raw.githubusercontent.com/marcodpt/tint/main/preact.html)
 - [hyperapp](https://raw.githubusercontent.com/marcodpt/tint/main/hyperapp.html)

Without changing absolutely anything in the model and layouts.

I'm open to listening to suggestions, solving bugs, and helping with any problems or features that the community deems necessary.

Sorry for opening an issue, but I didn't know a better way to expose my work which I believe is extremely useful for working with frameworks based on hyperscript.

Keep up the wonderful work you guys do with the mithril.js!

**Comment by kevinfiol** [comment]
@marcodpt I agree that opening an issue here is probably not the appropriate place for this. I'd recommend you open a show-n-tell thread over in the #general stream of the Mithril Zulip Chat: https://mithril.zulipchat.com/

**Comment by dead-claudia** [comment]
I'll just convert this into a discussion.

Edit: done (and put in the right category)
